### PR TITLE
Pi-hole FTL v5.10.2

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:**

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/FTL/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 

## 10

---

Update embedded `dnsmasq` version to 2.87test3 including a fix we submitted upstream to fix https://github.com/pi-hole/FTL/issues/1181. We also fix an issue with `REPLY_ADDR{4,6}` not being used for replies to `pi.hole`.

### CODE REVIEW

**Review by yubiuser — APPROVED** [review]
Tested both changes: `--rev-server` ending with non zero and `REPLY_ADDR4` are working

File: src/dnsmasq_interface.c [hunk 1]
```diff
@@ -800,6 +800,7 @@ void _FTL_iface(struct irec *recviface, const char *file, const int line)
 			char buffer[ADDRSTRLEN+1] = { 0 };
 			inet_ntop(AF_INET, &next_iface.addr4, buffer, ADDRSTRLEN);
 			logg("Configuration overwrites IPv4 address: %s", buffer);
+			next_iface.haveIPv4 = true;
```
Comment (yubiuser) [inline src/dnsmasq_interface.c:803]: This needs to go 6 lines up I think. (thanks to @jpgpi)

**Comment by jpgpi250** [comment]
please don't release yet

When the debug flags are set, it works
when the debug flags are removed, it stops working

confirmed by @yubiuser 

he says:

```
Guess it is because the added line is within an if(config.debug & DEBUG_NETWORKING) condition.
```

File: src/dnsmasq_interface.c (hunk 1 above)
Comment (DL6ER) [reply src/dnsmasq_interface.c]: Yes, I see I pushed the wrong commit (this was from earlier testing) but the tests complained anyway so nothing would have happened :-)

Fixed now. Binaries are currently building.

Comment (yubiuser) [reply src/dnsmasq_interface.c]: Working with the new builds even without the debug flags

Comment (jpgpi250) [reply src/dnsmasq_interface.c]: version  is now vDev-ccf8d15, tried again, appears to be working.
Thanks.

**Comment by dschaper** [comment]
I like squash.

**Review by dschaper — APPROVED** [review]

**Pull request merged**